src/main/java/artifacts/trinkets/TrinketsHelper.java

```diff
@@ -49,15 +51,17 @@ public static boolean isEquipped(Predicate<ItemStack> filter, LivingEntity entit
 
 		// LivingEntity not currently supported by Trinkets
 		if (entity instanceof PlayerEntity) {
-			Inventory inventory = TrinketsApi.getTrinketsInventory((PlayerEntity) entity);
+			Optional<TrinketComponent> maybeComponent = TrinketsApi.getTrinketComponent(entity);
 
-			for (int i = 0; i < inventory.size(); i++) {
-				ItemStack stack = inventory.getStack(i);
+			if (maybeComponent.isPresent()) {
+				TrinketComponent tComponent = maybeComponent.get();
 
-				if (!stack.isEmpty() && stack.getItem() instanceof TrinketArtifactItem
-						&& (Components.ARTIFACT_ENABLED.get(stack).get() || ignoreEffectsDisabled)) {
-					stacks.add(stack);
-				}
+				tComponent.getAllEquipped().forEach(pair -> {
```

Comment:
Now that trinkets provides this getAllEquipped method we can use a stream with filters to get to our list of stacks.